frontend/src/metabase/query_builder/selectors.js

```diff
@@ -244,15 +244,21 @@ export const getRawSeries = createSelector(
     // BUT the last executed "dataset_query" (to ensure data matches the query)
     return (
       results &&
-      question.atomicQueries().map((metricQuery, index) => ({
-        card: {
-          ...question.card(),
-          display: display,
-          visualization_settings: settings,
-          dataset_query: lastRunDatasetQuery,
-        },
-        data: results[index] && results[index].data,
-      }))
+      question.atomicQueries().map((metricQuery, index) => {
+        const { data, expected_timezone, actual_timezone } =
+          results[index] || {};
+        return {
+          card: {
+            ...question.card(),
+            display: display,
+            visualization_settings: settings,
+            dataset_query: lastRunDatasetQuery,
+            expected_timezone,
```

Comment:
I'm strong 👎 on copying the timezone to the `card` definition, or really requiring the frontend copy it from the response to anywhere else in `card`/`data`/`series`. This is only one of several places where queries are run (e.x. dashboards, QuestionAndResultLoader, probably others). I haven't tried it but it's unlikely this will work correctly anywhere outside of the query builder.

`Visualization` component takes the `series` array with objects containing `card` definition (a.k.a. Question, primarily for `display` and `visualization_settings`) and a `data` result set. We shouldn't require any processing of `card` and `data` obtained from the backend (aside from zippering them into `series`). The only reason we do in the code above is because we allow the user to modify the `display` and `visualization_settings` in the query builder.

This is also one of the reasons I was advocating for putting the timezone info on the column itself: even if it's always going to be the same across all columns in the query, we *use* that information on a column level, so it's easiest to ensure it's passed all the way down where it's used by including it on the column. I'd also be ok with it being in the `data` object though.

Actually, now that I think about it, with dashboard multiseries there's a possibility you could try to display two queries that were run in different timezones. We'd probably want to just pick the first timezone and show a warning.  See comments in the code below.